config/initializers/devise.rb

```diff
@@ -273,7 +273,7 @@
   # config.navigational_formats = ['*/*', :html]
 
   # The default HTTP method used to sign out a resource. Default is :delete.
-  config.sign_out_via = :delete
+  config.sign_out_via = :get
```

Comment:
We decided against changing the config to fix the sign out in #1954, although mainly out of an assumption (unresearched) about conventions. I'm not opposed to doing it now, but we could get around this by making it a form like in `app/views/activities/confirm_destroy.html.haml:20` and `app/views/shared/implementing_organisations/_table.html.haml:23`

Reply:
Urgh, didn’t realise this had a history. I’ll revisit tomorrow.

thanks!

Reply:
Yeah, I should've dug around and referenced it on the card! I don't have much of an opinion whether one approach is better than the other though - as I say, it was just based on an assumption about conventions more than anything

Reply:
Oh there are more instances than I realised! Looks like they're mostly `delete` but there's one `post`

Still, I'm not seeing anything we can't replicate with forms (and keep the same result in the UI, so no design needed). Obviously we can set the method in forms, but for the `confirm` functionality we can just add `html: { onsubmit: "return confirm('Are you sure?');" }`

If we do keep the hacked links, we should probably check that the tests are failable - without `js: true`, the sign out test was producing a false positive, hence why we didn't catch the bug for a while!

Reply:
Yeah, I was more thinking about how we can get rid of UJS. `onsubmit` would do the same thing, so it's not a regression in that sense, but if we make it a form users would still be able to take those actions without JS, just minus the confirmation dialog (the `onsubmit` is just ignored in that case). Arguably a risk to allow those actions without confirmation though?

Reply:
Do we want to get rid off UJS? 

Are you proposing that we:
- Remove UJS
- Replace all `link_to method:` with forms
- Replace all `link_to: confirm` with `onsubmit`

🤔

Reply:
Thought about this over lunch. Going to be stubborn and say:
- The guidance says we should have a separate page for confirmations (I think), which would be a larger piece of work for later
- Removing UJS and replacing with other JS doesn't seem worth it
- Seems like a bad time to do this in case we break something
- Replacing links/buttons with forms is going to give us styling problems
- This is low priority because internal users should probably just have JS enabled
- We might as well fix the sign out link while we are here because Devise give us the option to

But happy to follow the general agreement reach by the group!